### PR TITLE
Ignore undefine, null, or empty value rules when generate style.

### LINKED ISSUE #899: [jss-plugin-syntax-global] TypeError: Cannot read property '@global' of undefined
__Expected behavior:__

No errors. 

__Describe the bug:__

I can't create a stylesheet with an undefined rule. This can happen in Material-UI where we create CSS directly from the theme. 

__Codesandbox link:__ 

```jsx
import jss from "jss";
import presetDefault from "jss-preset-default";

jss.setup(presetDefault());
const s = jss.createStyleSheet({
  tutu: undefined
});
s.attach();
export default jss;
```

https://codesandbox.io/s/lyz1kryzl9

__Versions (please complete the following information):__
 - jss: 9.8.7

### PR DESCRIPTION
**Pull request by sammi**
Fix #899

### CODE REVIEW

File: packages/jss-plugin-syntax-global/src/index.test.js [hunk 1]
```diff
@@ -16,6 +16,25 @@ describe('jss-global', () => {
     jss = create(settings).use(global())
   })
 
+  describe('@global rule with null, undefined or empty value', () => {
```
Comment (HenriBeck) [inline packages/jss-plugin-syntax-global/src/index.test.js]: Can you also test `@global a` with `undefined` rule?

Comment (sammi) [reply packages/jss-plugin-syntax-global/src/index.test.js]: I guess b: undefined should cover it?

Comment (kof) [reply packages/jss-plugin-syntax-global/src/index.test.js]: What @HenriBeck describes is  a prefixed rule, what you are testing is a container rule:
```js
{
  '@global a': {color: 'red'}
}
```

Comment (sammi) [reply packages/jss-plugin-syntax-global/src/index.test.js]: Test it on sanbox, it's working.  

export default {
  '@global a': {color: 'red'}
};

Although not existing issue, I added it as a regression test in the PR.

Comment (sammi) [reply packages/jss-plugin-syntax-global/src/index.test.js]: oh, wait. @HenriBeck , what the "prefixed rule" test case looks like?

Comment (kof) [reply packages/jss-plugin-syntax-global/src/index.test.js]:
```js
'@global a': undefined
'@global b': null,
```

it might be already working, just to make sure we cover that case.

Comment (sammi) [reply packages/jss-plugin-syntax-global/src/index.test.js]: it is working. I added those test cases as well to protect more such empty value edge cases.

**Comment by DurgaPammi** [comment]
how can we right the JSS for the below example :

<Collapsible trigger="Start here">
    <p>This is the collapsible content. It can be any element or React component you like.</p>
  </Collapsible>
CSS is:

.Collapsible__trigger {
display: block;
font-weight: 600;
text-decoration: none;
color: #ffffff;
position: relative;
border: 1px solid white;
padding: 10px;
background: #f3f4f4;
color: black;
}
.Collapsible__trigger:after {
font-family: 'FontAwesome';
content: '\f107';
position: absolute;
right: 10px;
top: 10px;
display: block;
transition: transform 300ms;
}
.Collapsible__trigger.is-open:after {
transform: rotateZ(180deg);
}
.Collapsible__trigger.is-disabled {
opacity: 0.5;
background-color: grey; }

Can You please any one help me about this
